### ISSUE TITLE
v0.9.3 doesn't work

### ISSUE THREAD

**Issue by YWatchman** [lae/ansible-role-netbox#97]
v0.9.3 doesn't have the check for if netbox >= 2.7.11 so migration fails.

**Comment by lae** [comment]
0.9.3 should work for up until 2.7.12 (and in fact should work with 2.8.x, I believe, but its features aren't fully supported). If you're having an issue deploying 2.7.12 with it, please post your task output/error.

For 2.8.x please use https://github.com/lae/ansible-role-netbox/releases/tag/v1.0.0rc1.

**Comment by YWatchman** [comment]
Unfortunately just used v1.0.0rc1 so I could finish the netbox update, have no logs. :/ Updating to 2.7.12 failed:

https://github.com/lae/ansible-role-netbox/blob/v0.9.3/templates/configuration.py.j2#L33

This line changed in 2.7.11.

**Comment by lae** [comment]
That line did indeed change in 2.7.11 but NetBox upstream still supports the old configuration until 2.9.x. See https://github.com/netbox-community/netbox/pull/4366/commits/3590ed378d161dd724fad2dc73ff56da746352f8#diff-a1b75c101e583ea1264c4c0c390db558R183

**Comment by YWatchman** [comment]
I don't understand why migration failed then for 2.7.12. It failed on a KeyError stating the webhooks / tasks key.

**Comment by lae** [comment]
**For future reference, please always post any debug/log output you have when you open an issue.** It may contain more information that you might at first gleam from it. In this case, the key warning (not `KeyError`) was just extra output from the Django script rather than the actual error itself.

I attempted a 2.7.12 deployment on v0.9.3. The issue was with one of NetBox' dependencies making a breaking change, which @tyler-8 reported at https://github.com/netbox-community/netbox/issues/4633.

This is fixed upstream on NetBox 2.8.4, but adding a pip constraint to ignore django-rq 2.3.1 forces NetBox to install django-rq 2.3.2, so I'm committing that so that earlier versions still work.

```
TASK [lae.netbox : Run database migrations for NetBox] *************************
fatal: [default]: FAILED! => {
    "changed": false,
    "cmd": "./manage.py migrate --noinput",
    "path": "/srv/netbox/current/venv-py3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
    "syspath": [
        "/tmp/ansible_django_manage_payload_lwli04fj/ansible_django_manage_payload.zip",
        "/usr/lib/python37.zip",
        "/usr/lib/python3.7",
        "/usr/lib/python3.7/lib-dynload",
        "/usr/local/lib/python3.7/dist-packages",
        "/usr/lib/python3/dist-packages"
    ]
}

MSG:


:stderr: /srv/netbox/releases/netbox-2.7.12/netbox/netbox/settings.py:187: UserWarning: The 'webhooks' REDIS configuration section has been renamed to 'tasks'. Please update your configuration as support for the old name will be removed in a future release.
  "The 'webhooks' REDIS configuration section has been renamed to 'tasks'. Please update your configuration as "
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/base.py", line 361, in execute
    self.check()
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/base.py", line 390, in check
    include_deployment_checks=include_deployment_checks,
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 65, in _run_checks
    issues.extend(super()._run_checks(**kwargs))
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/management/base.py", line 377, in _run_checks
    return checks.run_checks(**kwargs)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/checks/registry.py", line 72, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/urls/resolvers.py", line 399, in check
    for pattern in self.url_patterns:
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/urls/resolvers.py", line 584, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/urls/resolvers.py", line 577, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/srv/netbox/releases/netbox-2.7.12/netbox/netbox/urls.py", line 77, in <module>
    path('admin/background-tasks/', include('django_rq.urls')),
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django_rq/urls.py", line 3, in <module>
    from . import views
  File "/srv/netbox/current/venv-py3/lib/python3.7/site-packages/django_rq/views.py", line 12, in <module>
    from rq.exceptions import NoSuchJobError, UnpickleError
ImportError: cannot import name 'UnpickleError' from 'rq.exceptions' (/srv/netbox/current/venv-py3/lib/python3.7/site-packages/rq/exceptions.py)


PLAY RECAP *********************************************************************
default                    : ok=79   changed=0    unreachable=0    failed=1    skipped=52   rescued=0    ignored=0
```

**Comment by npalladium** [comment]
This error occurs on Ubuntu 18.04 with Python3.8 even off master. It was fixed by changing netbox version to 2.8.6 where I believe the issue has been fixed upstream.
